src/main/java/com/google/sps/data/DataGraph.java

```diff
@@ -0,0 +1,125 @@
+package com.google.sps.data;
+
+// import java.util.ArrayList;
+// import java.util.HashMap;
+// import java.util.HashSet;
+// import java.util.List;
+// import java.util.Map;
+
+// import com.google.appengine.repackaged.com.google.protobuf.Struct;
+// import com.google.common.graph.*;
+// import com.google.common.graph.Graph;
+
+// import com.google.common.graph.MutableGraph;
+// import com.proto.GraphProtos.Node;
+
+// import com.google.sps.data.GraphNode;
+// import com.proto.GraphProtos.*;
+
+import com.google.common.graph.*;
+import com.proto.GraphProtos.Graph;
+import com.proto.GraphProtos.Node;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.protobuf.Struct;
+import java.util.Map;
+import java.util.HashSet;
+
+public class DataGraph {
```

Comment:
Why not use AutoValue? :)

Reply:
My understanding may be incorrect, but I thought the parameters passed into the constructor were also the private instance variables stored in AutoValue! And that's not what we want, right?

Reply:
It cannot be fully private, but could be package-visible only. I think that is sufficient for encapsulation.

https://github.com/google/auto/blob/master/value/userguide/howto.md#-keep-my-accessor-methods-private

The pros is to avoid boilerplates (equals/hashcode/getter/setter) and mistakes. IDE can certainly generate and maintain these boilerplates but then it is still visible to readers.